src/main/java/com/gotocompany/depot/maxcompute/schema/partition/TimestampPartitioningStrategy.java

```diff
@@ -0,0 +1,56 @@
+package com.gotocompany.depot.maxcompute.schema.partition;
+
+import com.aliyun.odps.Column;
+import com.aliyun.odps.PartitionSpec;
+import com.aliyun.odps.data.Record;
+import com.aliyun.odps.expression.TruncTime;
+import com.aliyun.odps.type.TypeInfoFactory;
+import com.gotocompany.depot.config.MaxComputeSinkConfig;
+import com.gotocompany.depot.maxcompute.schema.MaxComputeSchemaCache;
+
+public class TimestampPartitioningStrategy implements PartitioningStrategy {
+
+    private static final String DAY = "DAY";
+
+    private final MaxComputeSinkConfig maxComputeSinkConfig;
+    private MaxComputeSchemaCache maxComputeSchemaCache;
+
+    public TimestampPartitioningStrategy(MaxComputeSinkConfig maxComputeSinkConfig) {
+        this.maxComputeSinkConfig = maxComputeSinkConfig;
+    }
+
+    @Override
+    public String getOriginalPartitionColumnName() {
+        return maxComputeSinkConfig.getTablePartitionKey();
+    }
+
+    @Override
+    public boolean shouldReplaceOriginalColumn() {
+        return false;
+    }
+
+    @Override
+    public Column getPartitionColumn() {
+        Column column = Column.newBuilder(maxComputeSinkConfig.getTablePartitionColumnName(), TypeInfoFactory.STRING)
+                .build();
+        column.setGenerateExpression(new TruncTime(maxComputeSinkConfig.getTablePartitionKey(), DAY));
+        return column;
+    }
+
+    @Override
+    public PartitionSpec getPartitionSpec(Object object) {
+        PartitionSpec partitionSpec = new PartitionSpec();
+        maxComputeSchemaCache.getMaxComputeSchema()
+                .getTableSchema()
+                .getPartitionColumns()
+                .forEach(partitionColumn -> partitionSpec.set(partitionColumn.getName(), partitionColumn.getGenerateExpression()
+                                .generate((Record) object)));
+        return partitionSpec;
+    }
```

Comment:
The partition generation assumes the object is always a Record and doesn't handle invalid timestamps or null values properly. Also makes unnecessary schema lookups for each partition generation.

Reply:
null values will be automatically mapped to `__NULL__` by SDK, i've tested this